### PR TITLE
test: Add K8s test to set quota for a pod

### LINKED ISSUE #980: Add k8s test to set quota for a pod
Add a k8s test that sets a quota for the total number of pods

### PR DESCRIPTION
**Pull request by GabyCT**
This will set a quota for the total number of pods.

Fixes #980

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/kubernetes/k8s-pod-quota.bats [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+@test "Pod quota" {
+	resource_name="pod-quota"
+	deployment_name="deploymenttest"
+
+	# Create the resourcequota
+	sudo -E kubectl create -f "${pod_config_dir}/pod-quota.yaml"
+
+	# View information about resourcequota
+	sudo -E kubectl get resourcequota "$resource_name" --output=yaml | grep 'pods: "2"'
```
[the remark this answers is not in the document]
Comment (GabyCT) [reply integration/kubernetes/k8s-pod-quota.bats]: @jodh-intel , the output is
```
pods: "2"
```

[the remark this answers is not in the document]
Comment (jodh-intel) [reply integration/kubernetes/k8s-pod-quota.bats]: ack.

**Comment by jodh-intel** [comment]
lgtm

File: integration/kubernetes/k8s-pod-quota.bats [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	# Create deployment
+	sudo -E kubectl create -f "${pod_config_dir}/pod-quota-deployment.yaml"
+
+	# View information about the deployment
+	sudo -E kubectl get deployment "$deployment_name" --output=yaml | grep "forbidden: exceeded quota"
```
[the remark this answers is not in the document]
Comment (GabyCT) [reply integration/kubernetes/k8s-pod-quota.bats]: @jodh-intel , the return code is `0`

File: integration/kubernetes/k8s-pod-quota.bats [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		pod_config_dir="${BATS_TEST_DIRNAME}/untrusted_workloads"
+	fi
+}
+
+@test "Pod quota" {
+	resource_name="pod-quota"
+	deployment_name="deploymenttest"
+
+	# Create the resourcequota
+	sudo -E kubectl create -f "${pod_config_dir}/pod-quota.yaml"
```
Comment (chavafg) [inline integration/kubernetes/k8s-pod-quota.bats]: `pod-quota.yaml` is actually not defining a pod, so I think `quota.yaml` or `resource-quota.yaml` would be a better name.

File: integration/kubernetes/untrusted_workloads/pod-quota-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: deploymenttest
+  annotations:
```
Comment (chavafg) [inline integration/kubernetes/untrusted_workloads/pod-quota-deployment.yaml]: these annotations should go in the pod metadata, after L21

**Comment by GabyCT** [comment]
@chavafg changes applied

**Pull request merged**